app/controllers/api/v0/posts_controller.rb

```diff
@@ -0,0 +1,33 @@
+module Api
+  module V0
+    class PostsController < Api::V0::BaseController
+      include PostsHelper
+
+      before_action only: :show do
+        require_access_token Api::OpenidConnect::Scope.find_by(name: "read")
+      end
+
+      before_action only: %i(create destroy) do
+        require_access_token Api::OpenidConnect::Scope.find_by(name: "read"),
+                             Api::OpenidConnect::Scope.find_by(name: "write")
+      end
+
+      def show
+        posts_services = PostService.new(id: params[:id], user: current_user)
+        posts_services.mark_user_notifications
```

Comment:
I wonder if we should parametrize that. Retrieving a post without marking the notification as read feels like a very valid usecase.